src/sysc/kernel/sc_simcontext.cpp

```diff
@@ -1761,10 +1761,16 @@ sc_start()
               SC_EXIT_ON_STARVATION );
 }
 
+static int g_exit = EXIT_SUCCESS;
+SC_API int
+sc_exit_code() {
+    return g_exit;
+}
 
 SC_API void
-sc_stop()
+sc_stop(int exit_code)
 {
+    if (g_exit!=EXIT_SUCCESS) g_exit = exit_code;
```

Comment:
Shouldn't the assignment only be allowed if `g_exit==EXIT_SUCCESS`?

Reply:
ops - good point well made - sorry !!!